File: src/test/java/edu/cornell/cs/apl/viaduct/imp/parsing/ParserTest.java
package edu.cornell.cs.apl.viaduct.imp.parsing;

import edu.cornell.cs.apl.viaduct.ExamplesProvider;
import java.io.File;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.ArgumentsSource;

class ParserTest {
  @ParameterizedTest
  @ArgumentsSource(ExamplesProvider.class)
  void testParse(File file) throws Exception {
    Parser.parse(SourceFile.from(file));
  }
}

File: src/test/kotlin/edu/cornell/cs/apl/viaduct/parsing/ParserTest.kt
package edu.cornell.cs.apl.viaduct.parsing

import edu.cornell.cs.apl.viaduct.ExampleFileProvider
import java.io.File
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.ArgumentsSource

internal class ParserTest {
    @ParameterizedTest
    @ArgumentsSource(ExampleFileProvider::class)
    fun `it parses`(file: File) {
        SourceFile.from(file).parse()
    }
}
